commands.go

```diff
@@ -207,6 +208,41 @@ func NewCommandDeriveKey(pathStr string) (*apdu.Command, error) {
 	), nil
 }
 
+// Export a key
+//	@param {p1}
+//		0x00: current key - returns the key that is currently loaded and ready for signing. Does not use derivation path
+//		0x01: derive - returns derived key
+//		0x02: derive and make current - returns derived key and also sets it to the current key
+//  @param {p2}
+//		0x00: return public and private key pair
+//		0x01: return only the public key
+//  @param {pathStr}
+//		Derivation path of format "m/x/x/x/x/x", e.g. "m/44'/0'/0'/0/0"
+func NewCommandExportKey(p1 uint8, p2 uint8, pathStr string) (*apdu.Command, error) {
+	
+	// Choose to derive based on the value of p1
+	data := new(bytes.Buffer)
+	if (p1 == 0x01 || p1 == 0x02) {
+		_, path, err := derivationpath.Decode(pathStr)
```

Comment:
depending on the direction of the path, we can derive the key from master, parent, or current key.
if we OR the p1 with that value we allow all the possible options. We can use the same logic used in `NewCommandDeriveKey`

Reply:
Not sure I follow. We are just defining a derivation path here, right? Isn't the rest of the optionality captured in `p1`?

Reply:
sorry I didn't add much context here.
As specified in the docs of EXPORT KEY, if p1 is 0x02 it says:
>P1 = 0x02 derives the path but also changes the current path of the card. The source for derivation can be set by OR’ing P1 with the constants defined in the DERIVE KEY command. This allows deriving from master, parent or current.

So you can `0x02` is using an absolute path, but `0x02 | P1DeriveKeyFromCurrent` and `0x02 | P1DeriveKeyFromParent` derive starting from current key or parent key.

Reply:
Ooh I missed that! Fixed